docs/documentation/cert-manager.md

```diff
@@ -0,0 +1,203 @@
+kind: Issuer
+metadata:
+  name: kubernetes-example
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kubernetes-example
+spec:
+  encodeUsagesInRequest: false
+  isCA: false
+  issuerRef:
+    name: kubernetes-example
+  secretName: tls-secret
+```
+
+Apart from these two resources, the Cert-Manager Dekorate extension will also configure, part of the Deployment, a volume mounted from the secret that contain the Cert-Manager generated files to allow the application to access them and to configure the HTTPS/TLS endpoint:
+
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetes-example
+spec:
+  replicas: 1
+  template:
+    spec:
+      containers:
+        - name: kubernetes-example
+          volumeMounts:
+            - mountPath: /etc/certs
+              name: volume-certs
+              readOnly: true
+      volumes:
+        - name: volume-certs
+          secret:
+            optional: false
+            secretName: tls-secret
+```
+
+#### Usage
+
+To let the application access the path mounted, we have then to update also the properties of the application (Quarkus, Spring Boot, ...) to use the files that are mounted in the folder "/etc/certs" from the secret.
+
+To see a practical working example, please go to [the Spring Boot with Cert-Manager](https://github.com/dekorateio/dekorate/tree/main/examples/spring-boot-with-certmanager-example) example which uses a PKCS keystore.
+
+#### Securing Resources
+
+When securing your resources, it's important to validate that the requests are coming from known host names. For this purpose, we can use the `dnsNames` property which is part of the certificate configuration. For example, by adding the following `dekorate.certificate.dnsNames` property (it's a comma separated list of strings):
+
+```
+dekorate.certificate.dnsNames=foo.bar.com
+```
+
+The certificate will only allow requests accessing the server host `foo.bar.com`. Remark: If the DNS Host name do not exist, then you will get an error.
+
+Note that the applications in Kubernetes can be publicly exposed using [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resources, for example:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: kubernetes-example
+spec:
+  rules:
+  - host: foo.bar.com
+    http:
+      paths:
+      - pathType: Prefix
+        path: "/"
+        backend:
+          service:
+            name: kubernetes-example
+            port:
+              number: 8080
+  tls:
+    - hosts:
+        - foo.bar.com
+      secretName: tls-secret # < cert-manager will store the created certificate in this secret.
+```
+
+In Dekorate, you can generate the above Ingress resource by simply adding the following key properties:
+```
+dekorate.kubernetes.host=foo.bar.com
+dekorate.kubernetes.expose=true
+TODO: Allow to configure TLS configuration in the Ingress: https://github.com/dekorateio/dekorate/issues/967
```

Comment:
Will the TODO be fixed part of this PR ?

Reply:
no

Reply:
I've already updated this part. There is no TODO now.